streaming/java/streaming-runtime/src/main/java/org/ray/streaming/runtime/core/resource/Resources.java

```diff
@@ -0,0 +1,134 @@
+package org.ray.streaming.runtime.core.resource;
+
+import com.google.common.base.MoreObjects;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.ray.api.id.UniqueId;
+import org.ray.streaming.runtime.config.master.ResourceConfig;
+
+/**
+ * Resource description of ResourceManager.
+ */
+public class Resources implements Serializable {
+
+  /**
+   * Available containers registered to ResourceManager.
+   */
+  private List<Container> registerContainers = new ArrayList<>();
+
+  /**
+   * Mapping of allocated container to slots.
+   */
+  private Map<ContainerID, List<Slot>> allocatingMap = new HashMap<>(16);
+
+  /**
+   * Number of slots per container.
+   */
+  private int slotNumPerContainer = 0;
+
+  /**
+   * Number of actors per container.
+   */
+  private int actorPerContainer = 0;
+
+  /**
+   * Number of actors that the current container has allocated.
+   */
+  private int currentContainerAllocatedActorNum = 0;
+
+  /**
+   * The container index currently being allocated.
+   */
+  private int currentContainerIndex = 0;
+
+  private int maxActorNumPerContainer;
+
+
+  public Resources(ResourceConfig resourceConfig) {
+    maxActorNumPerContainer = resourceConfig.maxActorNumPerContainer();
+  }
+
+  public List<Container> getRegisterContainers() {
+    return registerContainers;
+  }
+
+  public void setSlotNumPerContainer(int slotNumPerContainer) {
+    this.slotNumPerContainer = slotNumPerContainer;
+  }
+
+  public int getSlotNumPerContainer() {
+    return slotNumPerContainer;
+  }
+
+  public void setRegisterContainers(
```

Comment:
Where do `setRegisterContainers ` comes from?

Reply:
```ResourceManager``` will start a thread detection ray cluster to register containers.